### PR TITLE
Edge support more platform

### PR DESCRIPTION
**Pull request by kalifun**
Currently cookies are stored in the same way as the Chrome browser stores them.

so, reuse chrome's code

Platform:

 - [x] Mac
 - [x] Win

### CODE REVIEW

**Comment by kalifun** [comment]
> This seems to make changes to Safari code too?
> 
> [Edit] I think that was changes from another PR? Could you rebase this?

Sorry, because Safari has been modified in other projects. So this part of the code is included in my code.

**Comment by kalifun** [comment]
> This seems to make changes to Safari code too?
> 
> [Edit] I think that was changes from another PR? Could you rebase this?

Hello, I have completed the base change. Can you help me see if there is any problem? @zellyn

File: filter.go [hunk 1]
```diff
@@ -188,34 +188,6 @@ func PathDepth(depth int) Filter {
...
-	return FilterFunc(func(cookie *Cookie) bool {
-		return cookie != nil && strings.HasSuffix(cookie.Value, suffix)
-	})
-}
-func ValueLen(length int) Filter {
-	return FilterFunc(func(cookie *Cookie) bool {
-		return cookie != nil && len(cookie.Value) == length
-	})
-}
-
```
Comment (srlehn) [inline filter.go:218]: Please remove the changes to filtering for now (this is an API change). I will try to add that delayed decryption without the removal of those filters (through some DelayedFilter type and type assertion or similar).

File: browser/safari/find_darwin.go [hunk 2]
```diff
@@ -7,10 +7,29 @@ import (
 	"path/filepath"
 )
 
+func fileExists(filename string) bool {
```
Comment (srlehn) [inline browser/safari/find_darwin.go:10]: Can you remove this file check? I think late failing when trying to open the file is better. the automatic search FindAllCookieStores will probably will deliver a few CookieStores. Then it can be filtered and only the remaining ones should be checked (simply opened). Oherwise profiles stored on the network will slow down.

File: browser/edge/find.go [hunk 3]
```diff
@@ -1,22 +1,13 @@
...
-	"errors"
-	"os"
-	"path/filepath"
-
 	"github.com/browserutils/kooky"
 	"github.com/browserutils/kooky/internal/chrome"
 	"github.com/browserutils/kooky/internal/chrome/find"
 	"github.com/browserutils/kooky/internal/cookies"
-	"github.com/browserutils/kooky/internal/ie"
-	_ "github.com/browserutils/kooky/internal/ie/find"
```
Comment (srlehn) [inline browser/edge/find.go:15]: leave the support for old edge  in. I would like to support analyzing old directory structures in the future.

File: internal/edge/find/find.go
Comment (srlehn) [inline internal/edge/find/find.go:1]: move these roots functions please into the internal/chrome/find package - that is meant for all chrome based browsers like current edge.

File: internal/chrome/chrome.go [hunk 4]
```diff
@@ -76,26 +76,15 @@ func (s *CookieStore) ReadCookies(filters ...kooky.Filter) ([]*kooky.Cookie, err
 			return err
 		}
 
-		encrypted_value, err := row.BytesStringOrFallback(`encrypted_value`, nil)
```
Comment (srlehn) [inline internal/chrome/chrome.go:79]: don't remove filter function. A rework of the filter order should be a separate PR and without API removal.

**Review by srlehn — CHANGES_REQUESTED** [review]

File: internal/edge/find/find.go
Comment (kalifun) [reply internal/edge/find/find.go:1]: Do you mean to follow the logic of chrome? Then add the path of edge to the detection of chrome? I feel that there will be something wrong with this.

File: internal/chrome/chrome.go (hunk 4 above)
Comment (kalifun) [reply internal/chrome/chrome.go:79]: He just encapsulated the function and did not remove this part of the logic. This part was modified by another author who forked this project, and I gave him the code. As a result, the code on both sides is unbalanced.

File: browser/safari/find_darwin.go (hunk 2 above)
Comment (kalifun) [reply browser/safari/find_darwin.go:10]: Maybe you need to see another pr, which he achieved.

File: filter.go (hunk 1 above)
Comment (kalifun) [reply filter.go:218]: If you don't think it's unreasonable, I can revise it based on the master of the current project. I'm really sorry, this part of the modification was submitted to another project.

**Comment by srlehn** [comment]
Finally merged via #75

Thanks a lot for the contribution!
